include/ozo/ext/std/duration.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <ozo/io/send.h>
+#include <ozo/io/recv.h>
+#include <ozo/type_traits.h>
+#include <boost/fusion/adapted/struct/define_struct.hpp>
+
+#include <chrono>
+#include <cstdint>
+
+BOOST_FUSION_DEFINE_STRUCT((ozo)(detail), pg_interval,
+    (std::int64_t, microseconds)
+    (std::int32_t, days)
+    (std::int32_t, months)
+)
+
+namespace ozo {
+
+/**
+ * @defgroup group-ext-std-chrono-duration-microseconds std::chrono::microseconds
+ * @ingroup group-ext-std
+ * @brief [std::chrono::microseconds](https://en.cppreference.com/w/cpp/chrono/duration) support
+ *
+ *@code
+#include <ozo/ext/std/duration.h>
+ *@endcode
+ *
+ * `std::chrono::microseconds` is mapped as `interval` PostgreSQL type.
+ */
+
+template <>
+struct send_impl<std::chrono::microseconds> {
+    template <typename OidMap>
+    static ostream& apply(ostream& out, const OidMap&, const std::chrono::microseconds& in) {
+        static_assert(ozo::OidMap<OidMap>, "OidMap should model ozo::OidMap");
+
+        using days = std::chrono::duration<std::int32_t, std::ratio<24 * std::chrono::hours::period::num>>;
+
+        write(out, detail::pg_interval {(in % days(1)).count(), std::chrono::duration_cast<days>(in).count(), 0});
+
+        return out;
+    }
+};
+
+template <>
+struct recv_impl<std::chrono::microseconds> {
+    template <typename OidMap>
+    static istream& apply(istream& in, size_type, const OidMap&, std::chrono::microseconds& out) {
+        static_assert(ozo::OidMap<OidMap>, "OidMap should model ozo::OidMap");
+
+        using microseconds = std::chrono::microseconds;
```

Comment:
BTW `using std::chrono` inside the function gives you more readable code here.

Reply:
Not sure. It brings `[std::chrono::]days` and `[std::chrono::]months` into local namespace, but it already contains "my" `days` and `months`.
I think it's enough to add `using std::chrono::duration_cast` to reach almost the same readability.